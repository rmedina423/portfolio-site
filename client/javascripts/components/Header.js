import React from 'react';

export default class Header extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      isMenuIconOpen: false
    }

    this.menuIconClick = this.menuIconClick.bind(this);
  }

  menuIconClick () {
    this.setState(prevState => ({
      isMenuIconOpen: !prevState.isMenuIconOpen
    }));
  }

  render () {
    return (
      <header>
        <div className="header-container">
          <h1 className="header-title">
            <a href="/">Ryan Medina</a>
          </h1>
          <div className="mobile-menu-block">
            <h2>Menu</h2>
            <MenuIcon isMenuIconOpen={this.state.isMenuIconOpen} menuIconClick={this.menuIconClick} />
          </div>
          <div className="desktop-menu-block">
            <div className="sub-menu-block">
              <a href="/">Portfolio</a>
              <a href="/">About & Contact</a>
            </div>
            <div className="sub-menu-block">
              <a href="/">Twitter</a>
              <a href="/">Instagram</a>
              <a href="/">Dribble</a>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

const MenuIcon = ({isMenuIconOpen, menuIconClick}) => {  
  return (
    <div className={isMenuIconOpen ? `menu-icon open` : `menu-icon`} onClick={menuIconClick}>
      <div className="bar-1"></div>
      <div className="bar-2"></div>
      <div className="bar-3"></div>
    </div>
  )
};