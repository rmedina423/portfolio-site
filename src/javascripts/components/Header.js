import React from 'react';

export default class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isMenuIconOpen: false
    };

    this.menuIconClick = this.menuIconClick.bind(this);
  }

  menuIconClick() {
    this.setState(prevState => ({
      isMenuIconOpen: !prevState.isMenuIconOpen
    }));
  }

  render() {
    return (
      <header>
        <div className="header-container">
          <h1 className="header-title">
            <a href="/">Ryan Medina</a>
          </h1>
          <div className="mobile-menu-block" onClick={this.menuIconClick}>
            <h2>Menu</h2>
            <MenuIcon isMenuIconOpen={this.state.isMenuIconOpen} />
          </div>
          <div className="desktop-menu-block">
            <div className="sub-menu-block">
              <a href="/">Portfolio</a>
              <a href="/">About & Contact</a>
            </div>
            <div className="sub-menu-block">
              <a href="/">Twitter</a>
              <a href="/">Instagram</a>
              <a href="/">Dribble</a>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

const MenuIcon = ({ isMenuIconOpen }) => (
  <div className={isMenuIconOpen ? 'menu-icon open' : 'menu-icon'}>
    <div className="bar-1" />
    <div className="bar-2" />
    <div className="bar-3" />
  </div>
);