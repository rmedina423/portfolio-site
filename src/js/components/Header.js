import React from 'react';
import { Link } from 'react-router-dom';

export default class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isMenuIconOpen: false
    };

    this.onMenuIconClick = this.onMenuIconClick.bind(this);
    this.onMenuIconKeyPress = this.onMenuIconKeyPress.bind(this);
  }

  onMenuIconClick() {
    this.setState(prevState => ({
      isMenuIconOpen: !prevState.isMenuIconOpen
    }));
  }

  onMenuIconKeyPress(event) {
    if (event.key === 'Enter') {
      this.setState(prevState => ({
        isMenuIconOpen: !prevState.isMenuIconOpen
      }));
    }
  }

  render() {
    return (
      <header>
        <div className="header-container">
          <h1 className="header-title">
            <Link to="/">Ryan Medina</Link>
          </h1>
          <div className="mobile-menu-block" onClick={this.onMenuIconClick} onKeyPress={this.onMenuIconKeyPress} role="button" tabIndex="0">
            <h2>Menu</h2>
            <div className={this.state.isMenuIconOpen ? 'menu-icon open' : 'menu-icon'}>
              <div className="bar-1" />
              <div className="bar-2" />
              <div className="bar-3" />
            </div>
          </div>
          <div className="desktop-menu-block">
            <div className="sub-menu-block">
              <Link to="/">Portfolio</Link>
              <Link to="/">About & Contact</Link>
            </div>
            <div className="sub-menu-block">
              <Link to="/">Twitter</Link>
              <Link to="/">Instagram</Link>
              <Link to="/">Dribble</Link>
            </div>
          </div>
        </div>
      </header>
    );
  }
}