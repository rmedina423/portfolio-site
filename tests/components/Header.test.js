import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../src/js/components/Header';

describe('Header Component', function () {
  beforeEach(() => {
    this.header = shallow(<Header />);
  });

  test('should match snapshot', () => {
    expect(this.header.getElement()).toMatchSnapshot();
  });

  test('changes state when .mobile-menu-block gets clicked', () => {
    expect(this.header.state('isMenuIconOpen')).toEqual(false);
    this.header.find('.mobile-menu-block').simulate('click');
    expect(this.header.state('isMenuIconOpen')).toEqual(true);
  });

  test('changes state when .mobile-menu-block gets a keyPress equal to `Enter`', () => {
    expect(this.header.state('isMenuIconOpen')).toEqual(false);
    this.header.find('.mobile-menu-block').simulate('keyPress', { key: 'Enter' });
    expect(this.header.state('isMenuIconOpen')).toEqual(true);
  });

  test('does NOT change state when .mobile-menu-block gets a keyPress NOT equal to `Enter`', () => {
    expect(this.header.state('isMenuIconOpen')).toEqual(false);
    this.header.find('.mobile-menu-block').simulate('keyPress', { key: 'NOT Enter' });
    expect(this.header.state('isMenuIconOpen')).toEqual(false);
  });
});