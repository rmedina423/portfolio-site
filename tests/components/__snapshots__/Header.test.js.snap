// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<header>
  <div
    className="header-container"
  >
    <h1
      className="header-title"
    >
      <Link
        replace={false}
        to="/"
      >
        Ryan Medina
      </Link>
    </h1>
    <div
      className="mobile-menu-block"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <h2>
        Menu
      </h2>
      <div
        className="menu-icon"
      >
        <div
          className="bar-1"
        />
        <div
          className="bar-2"
        />
        <div
          className="bar-3"
        />
      </div>
    </div>
    <div
      className="desktop-menu-block"
    >
      <div
        className="sub-menu-block"
      >
        <Link
          replace={false}
          to="/"
        >
          Portfolio
        </Link>
        <Link
          replace={false}
          to="/"
        >
          About & Contact
        </Link>
      </div>
      <div
        className="sub-menu-block"
      >
        <Link
          replace={false}
          to="/"
        >
          Twitter
        </Link>
        <Link
          replace={false}
          to="/"
        >
          Instagram
        </Link>
        <Link
          replace={false}
          to="/"
        >
          Dribble
        </Link>
      </div>
    </div>
  </div>
</header>
`;
